### PR TITLE
Speedup static build by utilizing CI cache on `/nix` folder

### PR DESCRIPTION
**Pull request by hswong3i**
Also see:
- https://github.com/containers/crun/pull/428
- https://github.com/containers/conmon/pull/189
- https://github.com/containers/skopeo/pull/973
- https://github.com/containers/buildah/pull/2428
- https://github.com/containers/podman/pull/7076
- https://github.com/cri-o/cri-o/pull/4012

Signed-off-by: Wong Hoi Sing Edison <hswong3i@gmail.com>

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -243,5 +248,6 @@ static_binary_task:
         mkdir -p /nix
         podman run --rm --privileged -ti -v /:/mnt nixos/nix cp -rfT /nix /mnt/nix
         podman run --rm --privileged -ti -v /nix:/nix -v ${PWD}:${PWD} -w ${PWD} nixos/nix nix --print-build-logs --option cores 8 --option max-jobs 8 build --file nix/
```
Comment (saschagrunert) [inline .cirrus.yml:250]: Do you think we can try to stick to nixos/nix directly without wrapping a virtual machine?

Comment (hswong3i) [reply .cirrus.yml:250]: I tried that multiple times before, but the key difficulty is: `nixos/nix` DON'T have bash.

If we are running with CI + docker directly, usually it assumed with bash exists, which make the test failed :-(

Comment (hswong3i) [reply .cirrus.yml:252]: P.S. I also try to create another nix-based container with Ubuntu 20.04 + bash (https://hub.docker.com/r/alvistack/nix) but this image generate another crazy behavior when integrating with CI :-(

**Comment by hswong3i** [comment]
Since https://github.com/cirruslabs/cirrus-ci-docs/issues/681 already get fixed, the static binary build phase should now speeding up, with a shared logic with https://github.com/containers/buildah/pull/2428 and https://github.com/containers/podman/pull/7076.

BTW, seems most CI test cases for conmon are now failing at the beginning, CI related to this PR couldn't be triggered...

**Comment by hswong3i** [comment]
@saschagrunert could we give a merge for this?

**Comment by saschagrunert** [comment]
I think we should somehow revive the CI first :D

**Comment by haircommander** [comment]
:sweat_smile: I will try to work with @cevich this week

**Pull request merged**